src/librustc_codegen_llvm/back/lto.rs

```diff
@@ -470,13 +479,26 @@ fn thin_lto(cgcx: &CodegenContext<LlvmCodegenBackend>,
 
         info!("thin LTO data created");
 
-        let import_map = if cgcx.incr_comp_session_dir.is_some() {
-            ThinLTOImports::from_thin_lto_data(data)
+        let (import_map_path, prev_import_map, curr_import_map) =
+            if let Some(ref incr_comp_session_dir) = cgcx.incr_comp_session_dir
+        {
+            let path = incr_comp_session_dir.join(THIN_LTO_IMPORTS_INCR_COMP_FILE_NAME);
+            let prev = if path.exists() {
+                // FIXME: can/should we recover from IO error occuring here
+                // (e.g. by clearing green_modules), rather than panicking from
+                // unwrap call?
+                Some(ThinLTOImports::load_from_file(&path).unwrap())
```

Comment:
We can recover by treating everything as changed somehow. Returning `None` here and making the code below handled that would be an option.

Reply:
You're right. That would probably be best.